manifests/init.pp

```diff
@@ -1,55 +1,57 @@
 # == Class: dhcp
 #
 class dhcp (
-  $dnsdomain            = undef,
-  $nameservers          = [ '8.8.8.8', '8.8.4.4' ],
-  $nameservers_ipv6     = [],
-  $ntpservers           = [],
-  $dnssearchdomains     = [],
-  $dhcp_conf_header     = 'INTERNAL_TEMPLATE',
-  $dhcp_conf_ddns       = 'INTERNAL_TEMPLATE',
-  $dhcp_conf_ntp        = 'INTERNAL_TEMPLATE',
-  $dhcp_conf_pxe        = 'INTERNAL_TEMPLATE',
-  $dhcp_conf_extra      = 'INTERNAL_TEMPLATE',
-  $dhcp_conf_fragments = {},
-  $interfaces           = undef,
-  $interface            = 'NOTSET',
-  $dnsupdatekey         = undef,
-  $ddns_update_style    = 'interim',
-  $dnskeyname           = undef,
-  $ddns_update_static   = 'on',
-  $ddns_update_optimize = 'on',
-  $pxeserver            = undef,
-  $pxefilename          = undef,
-  $mtu                  = undef,
-  $ipxe_filename        = undef,
-  $ipxe_bootstrap       = undef,
-  $logfacility          = 'daemon',
-  $default_lease_time   = 43200,
-  $max_lease_time       = 86400,
-  $service_ensure       = running,
-  $globaloptions        = '',
-  Optional[Integer[0,65535]] $omapi_port = undef,
-  Optional[String] $omapi_name = undef,
-  String $omapi_algorithm = 'HMAC-MD5',
-  Optional[String] $omapi_key = undef,
-  $authoritative        = true,
-  $extra_config         = '',
-  $dhcp_dir             = $dhcp::params::dhcp_dir,
-  $dhcpd_conf_filename  = 'dhcpd.conf',
-  $packagename          = $dhcp::params::packagename,
-  $servicename          = $dhcp::params::servicename,
-  $package_provider     = $dhcp::params::package_provider,
-  $ldap_port            = 389,
-  $ldap_server          = 'localhost',
-  $ldap_username        = 'cn=root, dc=example, dc=com',
-  $ldap_password        = '',
-  $ldap_base_dn         = 'dc=example, dc=com',
-  $ldap_method          = 'dynamic',
-  $ldap_debug_file      = undef,
-  $use_ldap             = false,
-  $option_code150_label = 'pxegrub',
-  $option_code150_value = 'text',
+  Optional[Array[String]] $dnsdomain                      = undef,
+  Array[Stdlib::Compat::Ipv4] $nameservers                = [ '8.8.8.8', '8.8.4.4' ],
+  # the ipv6 regex is currently wrong so we can't use it here
```

Comment:
If there is a PR for this, please link it. Makes it easier to check if it has been fixed and if so, which version.